MovieAggregator/Modules/MovieDetail/Controllers/MovieDetailViewController.swift

```diff
@@ -0,0 +1,241 @@
+// MovieDetailViewController.swift
+// Copyright © RoadMap. All rights reserved.
+
+import UIKit
+
+/// Screen with detail of selected movie.
+final class MovieDetailViewController: UIViewController {
+    // MARK: - Private Constants.
+
+    private enum Identifier {
+        static let posterCellID = "PosterCell"
+        static let infoCellID = "InfoCell"
+        static let overviewCellID = "OverviewCell"
+        static let genreCellID = "GenreCell"
+        static let castCellID = "CastCell"
+    }
+
+    private enum Constants {
+        static let backButtonImageName = "chevron.backward"
+        static let bottomShadowImageName = "bottomShadow"
+        static let castUrlComponent = "/credits"
+    }
+
+    private enum TypeOfCell {
+        case poster
+        case info
+        case genre
+        case overview
+        case actors
+    }
+
+    // MARK: - Public properties.
+
+    var movie: MovieDetail?
+    var urlMovieID = String()
+
+    // MARK: - Private properties.
+
+    private let typesOfCell: [TypeOfCell] = [.poster, .info, .genre, .overview, .actors]
+
+    // MARK: - Private visual components.
+
+    private lazy var movieTableView: UITableView = {
+        let tableView = UITableView()
+        tableView.separatorStyle = .none
+        tableView.scrollsToTop = false
+        tableView.allowsSelection = false
+        tableView.delegate = self
+        tableView.dataSource = self
+        tableView.translatesAutoresizingMaskIntoConstraints = false
+        return tableView
+    }()
+
+    private lazy var backButton: UIButton = {
+        let button = UIButton()
+        let config = UIImage.SymbolConfiguration(pointSize: 32, weight: .medium, scale: .default)
+        let image = UIImage(systemName: Constants.backButtonImageName, withConfiguration: config)
+        button.addTarget(self, action: #selector(backAction), for: .touchUpInside)
+        button.setImage(image, for: .normal)
+        button.tintColor = UIColor(named: Colors.favoriteMovieColorName)
+        button.translatesAutoresizingMaskIntoConstraints = false
+        return button
+    }()
+
+    // MARK: - Life cycle.
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupUI()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        configureNavigationBar(isHidden: true)
+    }
+
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        configureNavigationBar(isHidden: false)
+    }
+
+    override func viewWillLayoutSubviews() {
```

Comment:
```suggestion
    override func viewWillLayoutSubviews() {
```
вызывается после  viewWillDisappear